### PR TITLE
Infer Mainframer version from Git tag.

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
This is part of #198, next step is configuring deploys from tag builds on Travis.

### CODE REVIEW

**Comment by artem-zinnatullin** [comment]
Closes #192

File: ci/mainframer_version.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+set -e
+# Runs against Cargo.toml in working directory.
+
+# Script detects if build is triggered by git tag and sets it as Mainframer version, otherwise no-op.
+
+if [ -z "$TRAVIS_TAG" ]; then
+    echo "non-tag build, using version from Cargo.toml."
```
Comment (arturdryomov) [inline ci/mainframer_version.sh]: `Non-tag`?

File: ci/mainframer_version.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        echo "Git tag should start with 'v', ie 'v3.1.4'."
+        exit 1
+    fi
+
+    NEW_VERSION=${TRAVIS_TAG#"v"}
+    OLD_VERSION="3.0.0-dev"
+
+    if [[ "$OSTYPE" == "linux-gnu" ]]; then
+        sed --in-place "s/$OLD_VERSION/$NEW_VERSION/g" Cargo.toml
+    elif [[ "$OSTYPE" == "darwin"* ]]; then # macOS.
```
Comment (arturdryomov) [inline ci/mainframer_version.sh:22]: LOL, everyone knows that Darwin == macOS. Right? Riiight???

**Review by arturdryomov — APPROVED** [review]

File: ci/mainframer_version.sh (hunk 2 above)
Comment (artem-zinnatullin) [reply ci/mainframer_version.sh:22]: lol suuuure

File: ci/mainframer_version.sh (hunk 1 above)
Comment (artem-zinnatullin) [reply ci/mainframer_version.sh]: kk

**Pull request merged**